File: tippspiel2-spring/src/main/kotlin/com/github/zemke/tippspiel2/persistence/model/Community.kt
package com.github.zemke.tippspiel2.persistence.model

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.sql.Timestamp
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToMany
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

@Entity
data class Community(

        @Id @GeneratedValue val id: Long?,
        @Column(unique = true) @NotBlank val name: String,
        @ManyToMany val users: List<User>,
        @CreationTimestamp @NotNull val created: Timestamp = Timestamp.from(Date().toInstant()),
        @UpdateTimestamp @NotNull val modified: Timestamp = Timestamp.from(Date().toInstant())
)

File: tippspiel2-ember/src/main/webapp/app/models/community.js
import DS from 'ember-data';
import attr from 'ember-data/attr';
import { buildValidations, validator } from 'ember-cp-validations';

const Validations = buildValidations({
  name: validator('presence', true),
  users: validator('length', { min: 2 }),
});

export default DS.Model.extend(Validations, {
  name: attr('string'),
  users: DS.hasMany('user'),
  created: attr('Date'),
  modified: attr('Date'),
});
